app/models/user.rb

```diff
@@ -47,6 +48,9 @@ class User
 
   before_save :downcase_email
   after_create :create_initial_notification_method
+  after_create do
+    MailChimpWorker.perform_async(id)
```

Comment:
as per @toomuchpete 's comment above, there should be a condition that checks if it was subscribed, and only then perform this worker.